### PR TITLE
feat: Add Eslint configuration

### PR DESCRIPTION
**Pull request by hoangSTS**
feat: Add Eslint configuration

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,15 +1,25 @@
 module.exports = {
   root: true,
-  extends: '@react-native-community',
+  extends: ['@react-native-community', 'plugin:@typescript-eslint/recommended-requiring-type-checking'],
```
Comment (ngochuyduong) [inline .eslintrc.js:3]: Update config same as project-typescript for consistency

**Review by ngochuyduong — CHANGES_REQUESTED** [review]

**Review by loc-nguyenthien — APPROVED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (hoangSTS) [reply .eslintrc.js:3]: Updated!

**Review by TranQuangPhi — COMMENTED** [review]
Please add "yarn test" to the pre-commit hook

File: .eslintignore [hunk 2]
```diff
@@ -1,4 +1,4 @@
 **/*{.,-}min.js
 coverage/**/*.js
 storybook/storyLoader.js
-storybook-static
+storybook-static
```
Comment (ThinhKimVo) [inline .eslintignore:4]: add bottom space

File: .eslintrc.js [hunk 3]
```diff
@@ -1,16 +1,30 @@
...
+    '@react-native-community',
+    'eslint:recommended',
+    'plugin:react/recommended',
+    'plugin:react-hooks/recommended',
+    'plugin:@typescript-eslint/eslint-recommended',
+    'plugin:@typescript-eslint/recommended',
+  ],
   parser: '@typescript-eslint/parser',
-  plugins: ['@typescript-eslint'],
+  plugins: ['@typescript-eslint', 'import', 'unused-imports'],
```
Comment (ThinhKimVo) [inline .eslintrc.js:12]: could you explain `import` and `unused-imports` plugins?

**Review by ngochuyduong — APPROVED** [review]

File: .eslintrc.js (hunk 3 above)
Comment (loc-nguyenthien) [reply .eslintrc.js:12]: `import` and `unused-imports` are also config for the other 2 projects, they include rules:
- Warn if there are cycle import
- Error if import statements are not at the beginning of file
- Error if there are duplicate imports
- Error if there are unused imports

**Pull request merged**